### PR TITLE
Add RS-485 Module support

### PR DESCRIPTION
**Pull request by hubmartin**
Example project https://github.com/hubmartin/bcf-usb-rs485-module/blob/master/app/application.c

### CODE REVIEW

File: bcl/inc/bc_module_rs485.h [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+//! @return true On success
+//! @return false On failure
+
+bool bc_module_rs485_async_read_start(bc_tick_t timeout);
+
+//! @brief Stop async reading
+//! @return true On success
+//! @return false On failure
+
+bool bc_module_rs485_async_read_stop();
```
Comment (hubpav) [inline bcl/inc/bc_module_rs485.h]: Missing void

File: bcl/src/bc_module_rs485.c [hunk 2]
```diff
@@ -0,0 +1,403 @@
+#include <bc_module_rs485.h>
+
+
+#define _BC_MODULE_RS485_I2C_UART_ADDRESS 0x4e
+#define _BC_MODULE_RS485_I2C_TLA2021_ADDRESS 0x48
+
+#define _BC_SC16IS7x0_REG_IER       0x01 << 3
```
Comment (hubpav) [inline bcl/src/bc_module_rs485.c]: Always use `(...)` for compound `#define` statements, also we use all lowercase in hex numbers

File: bcl/src/bc_sc16is740.c [hunk 3]
```diff
@@ -198,3 +198,38 @@ size_t bc_sc16is740_read(bc_sc16is740_t *self, uint8_t *buffer, size_t length, b
...
+    {
+        return false;
+    }
+
+    if (!bc_i2c_memory_write_8b(self->_i2c_channel, self->_i2c_address, _BC_SC16IS740_SPECIAL_REG_DLH, 0x00))
+    {
+        return false;
+    }
+
+    // restore LCR
```
Comment (hubpav) [inline bcl/src/bc_sc16is740.c]: Start with uppercase letter

File: bcl/src/bc_module_rs485.c [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+}
+
+size_t bc_module_rs485_async_write(uint8_t *buffer, size_t length)
+{
+    if (!_bc_module_rs485._initialized || _bc_module_rs485._write_fifo == NULL)
+    {
+        return 0;
+    }
+
+    size_t bytes_written = bc_fifo_write(_bc_module_rs485._write_fifo, (uint8_t *)buffer, length);
```
Comment (hubpav) [inline bcl/src/bc_module_rs485.c]: Should be `(uint8_t *) buffer`

File: bcl/src/bc_module_rs485.c [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+        _bc_module_rs485._event_handler(BC_MODULE_RS485_EVENT_ASYNC_READ_TIMEOUT, _bc_module_rs485._event_param);
+    }
+
+    bc_scheduler_plan_current_relative(_bc_module_rs485._async_read_timeout);
+}
+
+static void _bc_module_rs485_task_interval(void *param)
+{
+    (void) param;
+    bc_module_rs485_measure();
```
Comment (hubpav) [inline bcl/src/bc_module_rs485.c]: Add empty line before "measure" call

File: bcl/src/bc_module_rs485.c [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+
+    size_t space_available;
+
+    if (bc_fifo_is_empty(_bc_module_rs485._write_fifo))
+    {
+        bc_scheduler_unregister(_bc_module_rs485._async_write_task_id);
+        _bc_module_rs485._async_write_in_progress = false;
+
+        _bc_module_rs485._event_handler(BC_MODULE_RS485_EVENT_ASYNC_WRITE_DONE, _bc_module_rs485._event_param);
+        return;
```
Comment (hubpav) [inline bcl/src/bc_module_rs485.c]: Add empty line before return

File: bcl/src/bc_module_rs485.c [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+    }
+
+    if (!bc_sc16is740_get_spaces_available(&_bc_module_rs485._sc16is750, &space_available))
+    {
+        bc_scheduler_unregister(_bc_module_rs485._async_write_task_id);
+        _bc_module_rs485._async_write_in_progress = false;
+
+        _bc_module_rs485._event_handler(BC_MODULE_RS485_EVENT_ERROR, _bc_module_rs485._event_param);
+        return;
```
Comment (hubpav) [inline bcl/src/bc_module_rs485.c]: Add empty line before return

File: bcl/src/bc_module_rs485.c [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+    }
+
+    _bc_module_rs485._async_read_timeout = timeout;
+    _bc_module_rs485._async_read_task_id = bc_scheduler_register(_bc_module_rs485_async_read_task, NULL, _bc_module_rs485._async_read_timeout);
+    _bc_module_rs485._async_read_in_progress = true;
+
+    return true;
+}
+
+bool bc_module_rs485_async_read_stop()
```
Comment (hubpav) [inline bcl/src/bc_module_rs485.c]: Missing void

File: bcl/inc/bc_module_rs485.h [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    //! @brief Error event
+    BC_MODULE_RS485_EVENT_ERROR = 0,
+
+    //! @brief Update event
+    BC_MODULE_RS485_EVENT_VOLTAGE = 1,
+
+    //! @brief Async write done
+    BC_MODULE_RS485_EVENT_ASYNC_WRITE_DONE = 2,
+
+    //! @brief Event is reading done
```
Comment (hubpav) [inline bcl/inc/bc_module_rs485.h]: Looking above on "Error event", it seems better convention as "Reading done event", so you can avoid "is" verb, hm?

**Comment by hubmartin** [comment]
@hubpav Fixed

**Comment by hubpav** [comment]
Thanks!

**Pull request merged**